src/client_side.cc

```diff
@@ -1253,7 +1253,10 @@ parseHttpRequest(ConnStateData *csd, const Http1::RequestParserPointer &hp)
             return NULL;
         }
 
-        if (csd->mayTunnelUnsupportedProto()) {
+        if (csd->ableToTunnelUnsupportedProto()) {
+            // The preserveDataForTunnelingUnsupportedProto is already checked
+            // inside Http::One::Server::parseOneRequest and the hp parser
+            // must preserved parsed data.
```

Comment:
This overly detailed comment is very likely to become stale (and hence confusing) soon IMHO. There is already a Must() inside the parsed() method that will trigger if this overall logic fails. And the reader can check what makes ableToTunnelUnsupportedProto() true. I think that is enough and suggest removing this comment.

If you insist on keeping the comment:
```suggestion
            // was told to preserve its parsed data.
```

I am also uncomfortable with the fact that this code is _not_ inside Http::One::Server -- it is not obvious that it does not apply to FTP requests.

Reply:
I removed the comment in commit e757c76

> I am also uncomfortable with the fact that this code is not inside Http::One::Server -- it is not obvious that it does not apply to FTP requests.

Not easy to move it here.
I do not know if we can support ftp protocol. Are we supporting transparent FTP proxying?

Reply:
> Are we supporting transparent FTP proxying?

Yes, AFAICT: `ftp_port` is explicitly documented to accept `intercept` and `tproxy` modes. We do not support on_unsupported_protocol for FTP though (yet?).

However, perhaps this is not important: I said that it is not obvious that this code does not apply to FTP requests (and, hence, should be moved to Http::One::Server), but it can be argued that it is obvious enough -- there are many HTTP-specific words/names in this context. Thus, I am OK with the code staying where it is if you think its current location works well.

Thank you for addressing my primary concern in this change request. I am resolving this change request.